### DISCUSSION TITLE
WhatsApp-like read receipts

### DISCUSSION THREAD

**Discussion by chetanyakan** [mattermost/community#94] [Enhancements]
Whatsapp-like read receipts that show who has seen the message.

![image](https://user-images.githubusercontent.com/35728906/123298646-5838cd80-d536-11eb-807e-8731244683b6.png)

![image](https://user-images.githubusercontent.com/35728906/123298673-5f5fdb80-d536-11eb-9592-4f597a955c50.png)

**Comment by esethna** [comment]
Related ticket! https://mattermost.atlassian.net/browse/MM-36529

**Reply by esethna** [reply]
I think we're going to start exploring the technical feasibility and working on a design for this, at least in DM/GMs where perf isn't as big of a concern

**Reply by darkLord19** [reply]
Also, I think it should be optional from the start as in user must have option to disable the read receipts from the first version itself to protect their privacy in case they don't want people knowing whether they read the message or not

**Reply by chetanyakan** [reply]
It might be interesting to have it similar to WhatsApp, where if you don't want other people to see whether you read their message or not, then you can't see if anyone has read your messages.

**Reply by esethna** [reply]
@darkLord19 totally agree, there would be an account setting and likely a feature introduction modal where the user could turn them off immediately if desired.

**Reply by esethna** [reply]
@chetanyakan yes it would be a setting that takes effect both ways, similar to WhatsApp or iMessages

**Comment by cnu1812** [comment]
Nice..!

**Comment by esethna** [comment]
Surprised this idea didn't get more upvotes. It's one of the most requested features on our feature idea forum!

**Reply by darkLord19** [reply]
Maybe because most of the people using mattermost will be more privacy focused than average and so this is not a big feature for them since mostly they'll turn it off. I'm just guessing but this is exactly what I think. It doesn't add any value for me and in addition I feel like it's intrusion in my personal space. That's why I wrote above that it should definitely have way to opt out from the inception of feature.

**Reply by Willyfrog** [reply]
i have this enabled in whatsapp and I don't usually think it gives me much value, most of the time people have their own setting disabled so I don't get the info about being read, and the most useful part of it is usually the "it was sent" which mattermost provides in other ways (not asking to retry)
